src/rate/index.tsx

```diff
@@ -0,0 +1,186 @@
+import { focus } from '@dojo/framework/core/middleware/focus';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import theme from '../middleware/theme';
+import * as css from '../theme/default/rate.m.css';
+import * as fixedCss from './rate.m.css';
+import * as baseCss from '../common/styles/base.m.css';
+import RadioGroup from '../radio-group';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import Icon from '../icon';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+
+export interface RateProperties {
+	/** Handler for when the value of the widget changes */
+	onValue?(value: number): void;
+	/* Number of icons to show, defaults to 5*/
+	max?: number;
+	/* Initial value for this widget */
+	initialValue?: number;
+	/* Controlled value for this widget */
+	value?: number;
+	/* The form name for this rate widget */
+	name?: string;
+	/* Flag to indicate if half stars should be used */
+	allowHalf?: boolean;
+	/* Flag to set the readonly state */
+	readOnly?: boolean;
+}
+
+export interface RateChildren {
+	label?: RenderResult;
```

Comment:
If we make a new parade release, jsdoc comments on these fields will be shown in a `children` table. Do we want to start adding them generally?

Reply:
good shout